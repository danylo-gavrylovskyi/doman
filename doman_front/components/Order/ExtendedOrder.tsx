import React from "react";

import { ExtendedOrderProps } from "./orderProps.interface";

import styles from "./order.module.scss";

export const ExtendedOrder = ({
	customer,
	orderId,
	totalPrice,
	createdAt,
	orderProducts,
}: ExtendedOrderProps) => {
	return (
		<section className={`${styles.minimizedOrder}`}>
			<header className={styles.orderHeader}>
				<div className={styles.info}>
					<span className={styles.bold}>№ {orderId}</span>
					<span>від {createdAt}</span>
				</div>

				<div>
					<svg width="25px" viewBox="0 0 24 24" fill="none">
						<path
							d="M17 15L12 10L7 15"
							stroke="rgb(0, 136, 204)"
							strokeWidth="1.5"
							strokeLinecap="round"
							strokeLinejoin="round"></path>
					</svg>
				</div>
			</header>

			<main className={styles.flexRowAlignCenter} style={{ alignItems: "flex-start" }}>
				<div className={styles.receiver}>
					<span className={styles.bold}>Отримувач замовлення</span>
					<span>
						{customer.firstName} {customer.lastName}
					</span>
					<span>{customer.phoneNumber}</span>
					<span>{customer.email}</span>
				</div>

				<div>
					{orderProducts.map((orderProduct) => {
						const { product } = orderProduct;
						return (
							<div
								key={product.id}
								className={`${styles.flexRowAlignCenter} ${styles.orderProduct}`}>
								<div style={{ display: "flex", width: "50%", alignItems: "center" }}>
									<img
										width="15%"
										src={`${process.env.NEXT_PUBLIC_API_URL}/uploads/productsImages/${product.image}`}
										alt={product.slug}></img>
									<div style={{ marginLeft: "5%" }}>{product.title}</div>
								</div>
								<div>
									{product.price}₴ x {orderProduct.quantity} од.
								</div>
								<div className={styles.bold}>{orderProduct.quantity * product.price} ₴</div>
								<div className={styles.separator}></div>
							</div>
						);
					})}

					<div className={styles.flexRowAlignCenter}>
						<span className={styles.totalPriceText}>Разом</span>
						<span className={styles.bold}>
							<b>{totalPrice}</b> ₴
						</span>
					</div>
				</div>
			</main>
		</section>
	);
};
