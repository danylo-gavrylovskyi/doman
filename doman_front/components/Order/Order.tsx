import React from "react";

import { OrderProps } from "./orderProps.interface";

import styles from "./order.module.scss";

export const Order = ({ orderId, totalPrice, createdAt, orderProducts, height }: OrderProps) => {
	return (
		<section
			style={{ height }}
			className={`${styles.minimizedOrder} ${styles.flexRowAlignCenter}`}>
			<div className={styles.sideInfo}>
				<span className={styles.bold}>№ {orderId}</span> від {createdAt}
			</div>

			<div className={styles.orderedProductsImages}>
				{orderProducts.map((orderProduct) => (
					<img
						key={orderProduct.id}
						width="100%"
						src={`${process.env.NEXT_PUBLIC_API_URL}/uploads/productsImages/${orderProduct.product.image}`}
						alt={orderProduct.product.slug}></img>
				))}
			</div>

			<div className={styles.flexRowAlignCenter}>
				<div style={{ marginRight: "10%" }} className={styles.sideInfo}>
					<span>До сплати</span>
					<span className={styles.bold}>{totalPrice} ₴</span>
				</div>
				<svg width="35px" viewBox="0 0 24 24" fill="none">
					<path
						d="M7 10L12 15L17 10"
						stroke="rgb(0, 136, 204)"
						strokeWidth="1.5"
						strokeLinecap="round"
						strokeLinejoin="round"></path>
				</svg>
			</div>
		</section>
	);
};
