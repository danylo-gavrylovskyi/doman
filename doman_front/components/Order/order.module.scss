.flexRowAlignCenter {
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
}

.minimizedOrder {
	justify-content: space-between;
	border: 1px solid rgba(0, 0, 0, 0.1);
	border-radius: 10px;
	padding: 1.5%;
	box-sizing: border-box;
	width: 100%;
	cursor: pointer;
}

.bold {
	font-size: 14px;
	font-weight: 700;
	color: black;

	b {
		font-size: 20px;
	}
}

.sideInfo {
	display: flex;
	flex-direction: column;
	font-size: 12px;
	color: #797878;
	white-space: nowrap;

	span:first-child {
		margin-bottom: 10%;
	}
}

.orderedProductsImages {
	width: 5%;
	display: flex;
	align-items: center;
}

.orderHeader {
	@extend .flexRowAlignCenter;
	width: 100%;
	font-size: 12px;
	margin-bottom: 2%;
}

.info {
	display: flex;
	align-items: center;
	white-space: nowrap;

	span:first-child {
		margin-right: 5%;
	}
}

.receiver {
	font-size: 13px;
	display: flex;
	flex-direction: column;
	margin-right: 5%;
}

.orderProduct {
	position: relative;
	padding: 1%;
	margin-bottom: 3%;
	box-sizing: border-box;
	font-size: 14px;
}

.separator {
	position: absolute;
	bottom: 0;
	right: 0;
	left: 0;
	border-bottom: 1px solid rgba(0, 0, 0, 0.1);
}

.totalPriceText {
	font-size: 14px;
	color: #797878;
}
