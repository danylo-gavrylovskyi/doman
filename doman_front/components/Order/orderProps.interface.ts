import { OrderProduct } from '@/types/order.interface';
import { User } from '@/types/user.interface';

export interface OrderProps {
	orderId: number;
	totalPrice: number;
	createdAt: string;
	orderProducts: OrderProduct[];
	height?: string;
}

export interface ExtendedOrderProps extends OrderProps {
	customer: Omit<User, 'password' | 'isAdmin'>;
}
