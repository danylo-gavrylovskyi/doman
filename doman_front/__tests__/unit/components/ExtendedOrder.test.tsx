import { render, screen } from "@testing-library/react"

import { ExtendedOrder } from "@/components/Order/ExtendedOrder"

import { mockExtendedOrderProps } from "@/__tests__/mocks/orderProps"

describe("ExtendedOrder (Unit)", () => {
    it("renders order details correctly", () => {
        render(<ExtendedOrder {...mockExtendedOrderProps} />)

        const { firstName, lastName, phoneNumber, email } = mockExtendedOrderProps.customer
        const pathToImage = `${process.env.NEXT_PUBLIC_API_URL}/uploads/productsImages/`

        expect(screen.getByText(`№ ${mockExtendedOrderProps.orderId}`)).toBeInTheDocument()
        expect(screen.getByText(`від ${mockExtendedOrderProps.createdAt}`)).toBeInTheDocument()
        expect(screen.getByText(`${mockExtendedOrderProps.totalPrice}`)).toBeInTheDocument()
        expect(screen.getByText("Отримувач замовлення")).toBeInTheDocument()
        expect(screen.getByText(`${firstName} ${lastName}`)).toBeInTheDocument()
        expect(screen.getByText(phoneNumber)).toBeInTheDocument()
        expect(screen.getByText(email)).toBeInTheDocument()
        mockExtendedOrderProps.orderProducts.forEach((orderProduct, i) => {
            const image = screen.getAllByRole("img")[i];
            expect(image).toHaveAttribute("src", `${pathToImage}${orderProduct.product.image}`)
            expect(image).toHaveAttribute("alt", orderProduct.product.slug)
            expect(screen.getByText(orderProduct.product.title)).toBeInTheDocument()
            expect(screen.getByText(`${orderProduct.product.price}₴ x ${orderProduct.quantity} од.`)).toBeInTheDocument()
            expect(screen.getByText(`${orderProduct.quantity * orderProduct.product.price} ₴`)).toBeInTheDocument()
        });
    })

    it("matches snapshot", () => {
        const { asFragment } = render(<ExtendedOrder {...mockExtendedOrderProps} />)
        expect(asFragment()).toMatchSnapshot()
    })
})