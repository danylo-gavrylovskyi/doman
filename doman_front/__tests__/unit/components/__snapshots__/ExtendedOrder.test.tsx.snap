// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtendedOrder (Unit) matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="minimizedOrder"
  >
    <header
      class="orderHeader"
    >
      <div
        class="info"
      >
        <span
          class="bold"
        >
          № 123
        </span>
        <span>
          від 2024-02-20
        </span>
      </div>
      <div>
        <svg
          fill="none"
          viewBox="0 0 24 24"
          width="25px"
        >
          <path
            d="M17 15L12 10L7 15"
            stroke="rgb(0, 136, 204)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </div>
    </header>
    <main
      class="flexRowAlignCenter"
      style="align-items: flex-start;"
    >
      <div
        class="receiver"
      >
        <span
          class="bold"
        >
          Отримувач замовлення
        </span>
        <span>
          John Doe
        </span>
        <span>
          123-456-789
        </span>
        <span>
          john.doe@example.com
        </span>
      </div>
      <div>
        <div
          class="flexRowAlignCenter orderProduct"
        >
          <div
            style="display: flex; width: 50%; align-items: center;"
          >
            <img
              alt="test-product-1"
              src="http://localhost:5000/uploads/productsImages/some-image-1.jpg"
              width="15%"
            />
            <div
              style="margin-left: 5%;"
            >
              Test Product 1
            </div>
          </div>
          <div>
            1200₴ x 2 од.
          </div>
          <div
            class="bold"
          >
            2400 ₴
          </div>
          <div
            class="separator"
          />
        </div>
        <div
          class="flexRowAlignCenter orderProduct"
        >
          <div
            style="display: flex; width: 50%; align-items: center;"
          >
            <img
              alt="test-product-2"
              src="http://localhost:5000/uploads/productsImages/some-image-2.jpg"
              width="15%"
            />
            <div
              style="margin-left: 5%;"
            >
              Test Product 2
            </div>
          </div>
          <div>
            1000₴ x 1 од.
          </div>
          <div
            class="bold"
          >
            1000 ₴
          </div>
          <div
            class="separator"
          />
        </div>
        <div
          class="flexRowAlignCenter"
        >
          <span
            class="totalPriceText"
          >
            Разом
          </span>
          <span
            class="bold"
          >
            <b>
              3400
            </b>
             ₴
          </span>
        </div>
      </div>
    </main>
  </section>
</DocumentFragment>
`;
